From: Juergen Gross <jgross@suse.com>
Date: Tue, 13 Jun 2017 14:18:32 +0200
Subject: xen/mce: don't issue error message for failed /dev/mcelog registration
Patch-mainline: 4.13-rc1
Git-commit: b867059018a5254cca14450eefb6fb8effa0f6dd
References: bnc#1036638

When running under Xen as dom0 /dev/mcelog is being registered by Xen
instead of the normal mcelog driver. Avoid an error message being
issued by the mcelog driver in this case. Instead issue an informative
message that Xen has registered the device.

Signed-off-by: Juergen Gross <jgross@suse.com>
---
 arch/x86/kernel/cpu/mcheck/mce.c |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

--- a/arch/x86/kernel/cpu/mcheck/mce.c
+++ b/arch/x86/kernel/cpu/mcheck/mce.c
@@ -2613,8 +2613,12 @@ static __init int mcheck_init_device(voi
 
 	/* register character device /dev/mcelog */
 	err = misc_register(&mce_chrdev_device);
-	if (err)
-		goto err_register;
+	if (err) {
+		if (err != -EBUSY)
+			goto err_register;
+		/* Xen dom0 might have registered the device already. */
+		pr_info("Unable to init device /dev/mcelog, already registered");
+	}
 
 	return 0;
 
